src/mon/FSCommands.cc

```diff
@@ -1074,3 +1055,30 @@ int FileSystemCommandHandler::_check_pool(
   return 0;
 }
 
+int FileSystemCommandHandler::is_op_allowed(
+    MonOpRequestRef op, FSMap &fsmap, const cmdmap_t& cmdmap,
+    std::stringstream &ss)
+{
+    string fs_name;
+    cmd_getval(cmdmap, "fs_name", fs_name);
+
+    fsmap.filter(op->get_session()->get_allowed_fsids());
+    auto fsids = op->get_session()->get_allowed_fsids();
```

Comment:
filtering is only necessary if in the `if` statement below? let's move it there.

Reply:
Rewrote this method to optimize it.